docs/swagger/swagger.yaml

```
@@ -584,451 +465,608 @@ components:
-    QuestionType:
-      type: string
-      example: Text
-      enum:
-        - Text
-        - TextArea
-        - Number
-        - MultipleChoice
-        - Checkbox
-        - LinearScale
-      description: |
-        どのタイプの質問か ("Text", "TextArea", "Number", "MultipleChoice", "Checkbox", "LinearScale")
+            - questionID
+            - created_at
+    Questions:
+      type: array
+      items:
+        $ref: "#/components/schemas/Question"
     QuestionBase:
       type: object
       properties:
-        page_num:
-          type: integer
-          example: 1
-          description: |
-            アンケートの何ページ目の質問か
-        question_num:
+        questionnaireID:
           type: integer
           example: 1
-          description: |
-            アンケートの質問のうち、何問目か
-        question_type:
-          $ref: '#/components/schemas/QuestionType'
-        body:
+        title:
+          type: string
+        description:
           type: string
-          example: 質問文
         is_required:
           type: boolean
-          example: true
-          description: |
-            回答必須かどうか
-        options:
-          type: array
-          items:
-            type: string
-            example: 選択肢1
-        scale_label_right:
-          type: string
-          example: そう思わない
-        scale_label_left:
-          type: string
-          example: そう思う
-        scale_min:
-          type: integer
-          example: 1
-        scale_max:
-          type: integer
-          example: 5
-        regex_pattern:
-          type: string
-          example: ''
-        min_bound:
-          type: string
-          example: ''
-        max_bound:
-          type: string
-          example: ''
       required:
-        - page_num
-        - question_num
-        - question_type
-        - body
+        - questionType
+        - title
+        - description
         - is_required
-        - options
-        - scale_label_right
-        - scale_label_left
-        - scale_min
-        - scale_max
-    NewQuestion:
+        - questionnaireID
+    QuestionSettingsByType:
+      oneOf:
+        - $ref: "#/components/schemas/QuestionSettingsText"
+        - $ref: "#/components/schemas/QuestionSettingsTextLong"
+        - $ref: "#/components/schemas/QuestionSettingsNumber"
+        - $ref: "#/components/schemas/QuestionSettingsSingleChoice"
+        - $ref: "#/components/schemas/QuestionSettingsMultipleChoice"
+        - $ref: "#/components/schemas/QuestionSettingsScale"
+    QuestionSettingsText:
       allOf:
-      - $ref: '#/components/schemas/QuestionBase'
-      - type: object
-        properties:
-          questionnaireID:
-            type: integer
-            example: 1
-        required:
-          - questionnaireID
-    Question:
+        - $ref: "#/components/schemas/QuestionTypeText"
+        - type: object
+          properties:
+            maxLength:
+              type: integer
+          required:
+            - maxLength
```

maxLengthはoptionalな方がいいかも
```suggestion
    QuestionSettingsText:
      allOf:
        - $ref: "#/components/schemas/QuestionTypeText"
        - type: object
          properties:
            maxLength:
              type: integer
```